web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -416,14 +582,16 @@ odoo.define('web_timeline.TimelineRenderer', function (require) {
             }
 
             var group = evt[self.last_group_bys[0]];
-            if (group && group instanceof Array) {
-                group = _.first(group);
+            if (group) {
+                if (group instanceof Array) {
+                    group = _.first(group);
+                }
             } else {
                 group = -1;
             }
-            _.each(self.colors, function (color) {
-                if (eval("'" + evt[color.field] + "' " + color.opt + " '" + color.value + "'")) {
-                    self.color = color.color;
+            _.each(self.colors, function (col) {
```

Comment:
Why are you changing this that is touched in the other PR?

Reply:
My bad, I thought I reverted it correctly, it has been adapted and I squashed the commits.